src/frontends/tensorflow_common/src/op/pack.cpp

```diff
@@ -16,22 +18,44 @@ namespace tensorflow {
 namespace op {
 
 OutputVector translate_pack_op(const NodeContext& node) {
-    default_op_checks(node, 1, {"Pack", "PACK"});
+    default_op_checks(node, 1, {"Pack", "PACK"}, true);
     auto num_size = static_cast<int>(node.get_input_size());
 
     auto axis = node.get_attribute<int64_t>("axis", 0);
+    auto input_shape = make_shared<v0::ShapeOf>(node.get_input(0));
+    auto rank = static_pointer_cast<v0::Constant>(input_shape->output(0).get_node_shared_ptr())->cast_vector<int64_t>().size();
```

Comment:
In general case, this rank cannot be computed  during model conversion. It can be computed only during run-time (not in this code).
Any other trick, how to adjust `axis` without `ShapeOf`. What value shoud be subtracted for correctness:
axis -= some_value ?
Please find some value for negative case

Reply:
@hibahassan1, any update?